sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1118,14 +1133,46 @@ evaluated as normal during command graph execution.
 [source,c++]
 ----
 using namespace ext::oneapi::experimental;
-auto node = graph.add([&](sycl::handler& cgh){
+auto node = graph.add([&](device, sycl::handler& cgh){
 // Host code here is evaluated during the call to add()
 cgh.host_task([=](){
   // Code here is evaluated as part of executing the command graph node
 });
 });
 ----
 
+=== Multi-device Graph
+
+A user can compose a graph with nodes targeting different devices. This allows
+the benefits of defining a execution graph ahead of submission to be extended
+to multi-device platforms. Without this capability a user would have to submit
+individual single device graphs and use events for dependencies, which is a
+usage model this extension is aiming to optimize. Automatic load balancing of
+commands across devices is not a problem this extension currently aims to solve,
+it is the responsibility of the user to decide the device each command will run
+on, not the SYCL runtime.
+
+A modifiable state `command_graph` contains nodes targeting specific devices,
+rather than being a device agnostic representation only tied to devices on
+finalization. This allows the implementation to process nodes which require
+device information when the command group function is evaluated. For example,
+SYCL reductions require a work-group/sub-group size to be known, which is
+normally gathered by the runtime from the device associated with the queue.
```

Comment:
I'm not sure if it's correct to say that this is _required_ by a reduction implementation, though in practice most likely every implementation would want to do this.

Reply:
Reworded to say an implementation *may* want this